### ISSUE TITLE
Clarify 64-bit scenario for Azure App Service

### ISSUE THREAD

**Issue by yahorsi** [aspnet/AspNetCore.Docs#13013]
Hi Guys,

The whole x64 bits topic is confusing.
These days people want to run in x64 mode just because we have some load and need some memory to be available. And we want to understand how is that possible.

I beleive there are different languages, and even worse different deployment "modes". And, every language developer needs to understand what he must do to deploy his app as x64.

What's currently written is completely confusing me, as .NET dev and architect. Just try to read it and try ti understabnd what does it means:

"
Platform
Runtimes for 64-bit (x64) and 32-bit (x86) apps are present on Azure App Service. The .NET Core SDK available on App Service is 32-bit, but you can deploy 64-bit apps using the Kudu console or via MSDeploy with a Visual Studio publish profile or CLI command.
"

So, what is runtimes available, what is runtime? Is that OS? Is that .NET runtime? Is that .NET Core runtime or anything else?

&gt;&gt;The .NET Core SDK available on App Service is 32-bit
Seriously? Just a word befoew you sad that 64 bits runtime is available, so, why .NET Core SDK is not? And what is Kudu console? Needs to be clarified, e.g. we use Azure Pipelines

And then, what the hell Platform-&gt;64 Bit setting does

So, from the one who builds .NET Core and needs to deploy stuff as x64 the whole topic looks like a confusion.

Sorry for the a bit upset tone, and thatnk you for letting us post the feedback!

Linking:
https://github.com/aspnet/AspNetCore.Docs/issues/9800

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a2eab542-a420-d6c5-d314-a420837cf602
* Version Independent ID: 48c12168-54f3-0b83-f136-565f483a03e9
* Content: [Deploy ASP.NET Core apps to Azure App Service](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/azure-apps/?view=aspnetcore-2.2)
* Content Source: [aspnetcore/host-and-deploy/azure-apps/index.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/host-and-deploy/azure-apps/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @yahorsi ...

* The .NET Core runtime is the execution environment for .NET Core apps. It's not the OS. The OS is part of the app's entire environment but not part of the .NET Core runtime. There are 32- and 64-bit runtimes.
* Azure App Service does have both the 32- and 64-bit runtimes available ever since .NET Core 2.2 was released. Note that to set the **Platform** to **64 Bit** in the App Service's **Configuration** > **General settings** that you need to have a *Basic* or higher service plan.
* The SDK is a set of libs and tools for building apps. It includes the CLI and the runtime. There are 32- and 64-bit SDKs.
* Kudu is a service in the Azure App Service. It's cross-linked there in that section, so you can read more about what it does by following that link. The main scenario for this issue is that there is a debug console in Kudu where you can access the 32-bit .NET Core SDK and run CLI commands. You can also drag-and-drop into the Kudu console to deploy an app to App Service.

So, I'll break that section down and explain it out a bit more and then make some remarks on what we'll do next with the content ...

> Runtimes for 64-bit (x64) and 32-bit (x86) apps are present on Azure App Service.

You can deploy a framework-dependent app, and it will run either 32-bit or 64-bit depending on how you have **Platform** set in the App Service **General settings**.

For example, I have a test app up there called "rp22," which is a 2.2 Razor Pages app. I dropped some Razor markup into its Index page so that I could show you the bitness of its process ...

```cshtml
<p>
    Is 64-bit process: @System.Environment.Is64BitProcess
</p>
```

This app is published using VS's built in **Publish** process, which uses a Publish Profile set up for App Service. I just used the built in VS UI to create the profile. It's a framework-dependent deployment, so it looks like this in VS ...

<img width="244" alt="Capture" src="https://user-images.githubusercontent.com/1622880/60198237-bd05a400-9806-11e9-9fbe-c60c628bb970.PNG">

Then, I publish and deploy the app.

When **Platform** is set to **32 Bit** ...

<img width="315" alt="64bitfalse" src="https://user-images.githubusercontent.com/1622880/60198695-d529f300-9807-11e9-9172-3e99368da9b3.PNG">

When **Platform** is changed to **64 Bit** ... 

<img width="296" alt="64bittrue" src="https://user-images.githubusercontent.com/1622880/60198698-d78c4d00-9807-11e9-93b0-8d227cbb936a.PNG">

In your scenario, the happy path is probably to publish/deploy a framework-dependent deployment (FDD) and set **Platform** as I show here. There are edge cases: self-contained deployments (SCD) and preview deployments. However, you seem like you don't have a special need for SCD or preview releases.

WRT Azure Pipelines, I don't use or know it in any detail, so I can't remark on it. If your pipeline produces an FDD and gets it into the right folder of the App Service, I don't think there are any differences as far as App Service is concerned. Set **Platform** to **64 Bit**, and it should ✨ *Just Work!*:tm: ✨ 🤞.

> The .NET Core SDK available on App Service is 32-bit, but you can deploy 64-bit apps using the Kudu console or via MSDeploy with a Visual Studio publish profile or CLI command.

You don't want to use the SDK that's available in the Kudu console because you're trying to build 64-bit apps. Engineering guidance is that if you desire 64-bit apps, use the 64-bit SDK. You'll be building your apps locally and in Azure Pipelines, so the fact that only the 32-bit SDK is available in Kudu shouldn't affect your deployment scenario.

What the last part of that is **_trying to say_** (and failing a bit on ☹️) is that you can drag-and-drop from a folder publish (contents of the *publish*) folder on your hard drive in your locally-published app **directly into the Kudu console** at *D:\home\site\wwwroot*. Yep ... you can just select the files and drag them right into the browser into the Kudu UI.

In your scenario, it's the last bit of that sentence that describes the approach that I think you want: Use the VS **Publish** process with the nice UI experience and Publish Profile that VS can create for you with just a few mouse clicks.

You asked some general .NET Core questions ... runtime & OS. That type of high-level architectural coverage starts in the [.NET Guide](https://docs.microsoft.com/dotnet/standard/) with related .NET Core-specific concepts in the following node, the [ASP.NET Core Guide](https://docs.microsoft.com/dotnet/core/). There's info on runtimes ... SDK ... etc. Check those docs out.

Finally, what to do here? For one thing, I went on a little doc hunt to see what bitness coverage exists over in .NET Core docs and Azure docs. I was quite **_underwhelmed_** by what I found. There are sporadic mentions, primarily pertaining to self-contained deployments and the Runtime IDentifier (RID) that specifies a particular bitness when publishing an SCD, but not a comprehensive, clear, helpful overview of all of the scenarios and what setting does what and when.

Leave this issue open. I'll confer with the Product Unit on this and we'll see how they would like to proceed.

**Comment by bwarren18** [comment]
I appreciate the additional information but this is still really confusing.  

I am trying to use a cmdlet that requires 64 bit but I am struggling to get it to work in Azure AppService.

I created a new AppService, chose a Windows Plan and chose the S1 Tier.  I tried with the Stack set to .NET Core and .NET (V4.7) and set the Platform to 64 bit. 
However in both cases [Environment]::Is64BitProcess returns false and the cmdlet (invoke-sqlcmd) fails.
